internal/security/secretstore/init.go

```diff
@@ -59,132 +62,159 @@ func NewBootstrap(insecureSkipVerify bool, vaultInterval int) *Bootstrap {
 	}
 
-	vaultProtocol := configuration.SecretService.Protocol
-	vaultHost := fmt.Sprintf("%s:%v", configuration.SecretService.Server, configuration.SecretService.Port)
 	intervalDuration := time.Duration(b.vaultInterval) * time.Second
-	vc := secretstoreclient.NewSecretStoreClient(lc, req, vaultProtocol, vaultHost)
+	clientConfig := types.SecretConfig{
+		Type:     secretStoreConfig.Type,
+		Protocol: secretStoreConfig.Protocol,
+		Host:     secretStoreConfig.Host,
+		Port:     secretStoreConfig.Port,
+	}
+	client, err := secrets.NewSecretStoreClient(clientConfig, lc, httpCaller)
+	if err != nil {
+		lc.Errorf("failed to create SecretStoreClient: %s", err.Error())
+		return false
+	}
+
+	lc.Info("SecretStoreClient created")
+
 	pipedHexReader := pipedhexreader.NewPipedHexReader()
-	kdf := kdf.NewKdf(fileOpener, configuration.SecretService.TokenFolderPath, sha256.New)
-	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, kdf)
+	keyDeriver := kdf.NewKdf(fileOpener, secretStoreConfig.TokenFolderPath, sha256.New)
+	vmkEncryption := NewVMKEncryption(fileOpener, pipedHexReader, keyDeriver)
 
 	hook := os.Getenv("IKM_HOOK")
 	if len(hook) > 0 {
 		err := vmkEncryption.LoadIKM(hook)
 		defer vmkEncryption.WipeIKM() // Ensure IKM is wiped from memory
 		if err != nil {
-			lc.Error(fmt.Sprintf("failed to setup vault master key encryption: %s", err.Error()))
+			lc.Errorf("failed to setup vault master key encryption: %s", err.Error())
 			return false
 		}
 		lc.Info("Enabled encryption of Vault master key")
 	} else {
 		lc.Info("vault master key encryption not enabled. IKM_HOOK not set.")
 	}
 
-	var initResponse secretstoreclient.InitResponse // reused many places in below flow
+	var initResponse types.InitResponse // reused many places in below flow
 
 	//step 3: initialize and unseal Vault
 	for shouldContinue := true; shouldContinue; {
 		// Anonymous function used to prevent file handles from accumulating
 		successful := func() bool {
-			sCode, _ := vc.HealthCheck()
+			sCode, _ := client.HealthCheck()
 
 			switch sCode {
 			case http.StatusOK:
 				// Load the init response from disk since we need it to regenerate root token later
-				if err := loadInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to load init response: %s", err.Error()))
+				if err := loadInitResponse(lc, fileOpener, secretStoreConfig, &initResponse); err != nil {
+					lc.Errorf("unable to load init response: %s", err.Error())
 					return false
 				}
-				lc.Info(fmt.Sprintf("vault is initialized and unsealed (status code: %d)", sCode))
+				lc.Infof("vault is initialized and unsealed (status code: %d)", sCode)
 				shouldContinue = false
+
 			case http.StatusTooManyRequests:
-				lc.Error(fmt.Sprintf("vault is unsealed and in standby mode (Status Code: %d)", sCode))
+				lc.Errorf("vault is unsealed and in standby mode (Status Code: %d)", sCode)
```

Comment:
IMO this is not an "error".  This is a carryover from the original implementation.  It will either go into ready mode or unseal.  Either way we're done here.

Reply:
Ok

Reply:
Replace with a `why` comment.